### PR TITLE
add commands to Dockerfile to set up MongoDB

### PR DESCRIPTION
**Pull request by OuyangDanwen**
Added MongoDB setup and clean up commands. You should be able to connect to it from host system at port 27017. Installation of MongoDB is required on the host system as well for it to work. Upon running, MongoDB becomes a background process.

### CODE REVIEW

**Comment by NgoKimPhu** [comment]
How do you hook this up to the Flask app? Are there built-in supports or we have to connect manually?

File: Dockerfile [hunk 1]
```diff
@@ -5,10 +5,23 @@ RUN apt-get install -y apache2
...
 RUN echo "ServerName localhost  " >> /etc/apache2/apache2.conf
 RUN echo "$user     hard    nproc       20" >> /etc/security/limits.conf
 ADD ./src/service /service
 ADD ./src/html /var/www/html
 EXPOSE 80
 EXPOSE 8080
+# Expose ports.
+#   - 27017: process
+#   - 28017: http
+EXPOSE 27017
```
Comment (NgoKimPhu) [inline Dockerfile]: Should we expose this? If mongodb runs within the container with the app itself it should already be visible to Flask

**Comment by OuyangDanwen** [comment]
What do you mean by manually? Python code can be used to do connections and all the transactions for all the endpoints later  inside the container. Exposing the port is for you to check the state of the DB from host. By right, DB should not be in the same docker container. So exposing the port should originally serve the purpose of inter-container communication.

**Comment by NgoKimPhu** [comment]
But you're exposing it from inside the same container, and by manually I mean without API support from Flask

**Comment by NgoKimPhu** [comment]
And by manually I mean without API support from Flask

**Review by NgoKimPhu — APPROVED** [review]
Expose approved for now

File: Dockerfile (hunk 1 above)
Comment (OuyangDanwen) [reply Dockerfile]: This is just for your own testing from host, I will add a comment to this.

**Comment by NgoKimPhu** [comment]
retesting

File: src/service/start_services.sh [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
 
 apachectl start
-python /service/app.py
+mongod --smallfiles --fork --logpath /log/mongod.log
+python ../service/app.py
```
Comment (NgoKimPhu) [inline src/service/start_services.sh]: Why the change to `../service/app.py`? See [Dockerfile#R19](files#diff-3254677a7917c6c01f55212f86c57fbfR19)
